### PR TITLE
Eclipse JKube Remote Development 🎉

### PR DESCRIPTION
**Pull request by manusa**
## Description

Code implementation for #508.

Documentation will be provided in a follow-up PR.

You can try it out  with my EclipseCon 2022 demo project: https://github.com/marcnuri-demo/202210-eclipsecon

Dependencies:
```
maven/mavencentral/org.apache.sshd/sshd-common/2.9.1, Apache-2.0 AND ISC, approved, #3224
maven/mavencentral/org.apache.sshd/sshd-core/2.9.1, Apache-2.0, approved, #3222
```

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [ ] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [ ] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] My code follows the style guidelines of this project
 - [ ] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [ ] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [ ] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/remote-dev/NOTES.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+mkfifo /tmp/fifo
+nc -l -p 6667 < /tmp/fifo | nc -u tftp-server 123 > /tmp/fifo
+
+### Connect from client (local machine)
+
+mkfifo /tmp/fifo
+nc -l -u -p 123 < /tmp/fifo | nc localhost 6667 > /tmp/fifo
+
+tftp localhost 6969
```
Comment (rohanKanojia) [inline jkube-kit/remote-dev/NOTES.md:31]: Is this required?

Comment (manusa) [reply jkube-kit/remote-dev/NOTES.md:31]: no, I need to move this elsewhere

**Review by rohanKanojia — APPROVED** [review]
:ok_hand:

**Comment by rohanKanojia** [comment]
I guess you'll add a quickstart for this along with the documentation in follow-up PR.

**Comment by manusa** [comment]
> I guess you'll add a quickstart for this along with the documentation in follow-up PR.

I will add documentation.
For quickstart we might have to discuss since all of the quickstarts are susceptible of dealing with this functionality.

File: jkube-kit/doc/src/main/asciidoc/inc/remote-dev/_index.adoc [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+[[jkube:remote-dev]]
+// TODO: Fix indentation problems due to misalignment with TOC headings
+ifeval::["{plugin-type}" == "maven"]
+== *{remote-dev-goal}* (preview)
+endif::[]
+ifeval::["{plugin-type}" == "gradle"]
+=== *{remote-dev-goal}* (preview)
+endif::[]
+
+The Eclipse JKube Remote Development goal allows you to run and debug code in your *local* machine:
```
Comment (rohanKanojia) [inline jkube-kit/doc/src/main/asciidoc/inc/remote-dev/_index.adoc:15]: Maybe this line should be moved in if else block
```suggestion
The Eclipse JKube Remote Development (goal|task) allows you to run and debug code in your *local* machine:
```

**Review by rohanKanojia — APPROVED** [review]

File: jkube-kit/doc/src/main/asciidoc/inc/remote-dev/_index.adoc (hunk 2 above)
Comment (manusa) [reply jkube-kit/doc/src/main/asciidoc/inc/remote-dev/_index.adoc:15]: I'll rephrase

**Pull request merged**